torch_xla/csrc/xla_lower_util.cpp

```diff
@@ -749,4 +749,79 @@ xla::XlaOp BuildMaskedScatter(xla::XlaOp input, xla::XlaOp mask,
       scatter_dnums);
 }
 
+std::vector<xla::XlaOp> BuildAmpForachNonFiniteCheckAndUnscale(
+    const std::vector<xla::XlaOp>& inputs, const xla::XlaOp& found_inf_float,
+    const xla::XlaOp& inv_scale) {
+  const xla::PrimitiveType origin_type =
+      XlaHelpers::ShapeOfXlaOp(found_inf_float).element_type();
+  std::vector<xla::XlaOp> found_infs;
+  xla::XlaOp one = xla::One(inputs[0].builder(), xla::PrimitiveType::S32);
+  for (size_t i = 0; i < inputs.size(); ++i) {
+    xla::XlaOp all_finite =
+        xla::ReduceAll(xla::ConvertElementType(xla::IsFinite(inputs[i]),
+                                               xla::PrimitiveType::S32),
+                       one,
+                       xla::CreateScalarAndComputation(xla::PrimitiveType::S32,
+                                                       inputs[i].builder()));
+    found_infs.push_back(one - all_finite);
+  }
+  xla::XlaOp found_inf =
+      xla::ConvertElementType(found_inf_float, xla::PrimitiveType::S32);
+  for (size_t i = 0; i < found_infs.size(); ++i) {
+    found_inf = xla::Or(found_inf, found_infs[i]);
+  }
```

Comment:
ditto, you can leave it as `xla::PrimitiveType::PRED`, no need to convert it to `S32`

Reply:
Done.